android/src/main/java/com/pspdfkit/react/PSPDFKitModule.java

```diff
@@ -13,26 +13,42 @@
 
 package com.pspdfkit.react;
 
+import android.app.Activity;
+import android.app.Application;
 import android.net.Uri;
+import android.os.Bundle;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.bridge.ReadableMap;
 import com.pspdfkit.PSPDFKit;
+import com.pspdfkit.document.PdfDocument;
+import com.pspdfkit.listeners.SimpleDocumentListener;
 import com.pspdfkit.ui.PdfActivity;
 
 import java.util.HashMap;
 import java.util.Map;
 
-public class PSPDFKitModule extends ReactContextBaseJavaModule {
+public class PSPDFKitModule extends ReactContextBaseJavaModule implements Application.ActivityLifecycleCallbacks {
 
     private static final String VERSION_KEY = "versionString";
     private static final String FILE_SCHEME = "file:///";
 
+    @Nullable
+    private Activity currentActivity;
+    @Nullable
+    private Runnable onPdfActivityOpenedTask;
+
     public PSPDFKitModule(ReactApplicationContext reactContext) {
         super(reactContext);
+        if (reactContext.getApplicationContext() instanceof Application) {
+            Application application = (Application) reactContext.getApplicationContext();
```

Comment:
This can lead to class cast exceptions, since some OEMs tend to wrap the application context (which means you won't be able to cast it to your application class).